opennms-dao/src/main/java/org/opennms/netmgt/dao/support/DefaultResourceDao.java

```diff
@@ -398,4 +401,33 @@ private boolean isDomainResourceTypeUsed() {
         }
         return false;
     }
+
+    @Override
+    public ResourceId getResourceId(CollectionResource resource, long nodeId) {
```

Comment:
Good candidate for a unit test.

Reply:
Added. Also validated generated `ResourceId` with Rest API.